src/ploigos_step_runner/step_implementers/undeploy/argocd_delete.py

```diff
@@ -0,0 +1,204 @@
+`service-name`              | Yes       |            | Used to build ArgoCD application name.
+"""
+
+import sys
+import sh
+from ploigos_step_runner.exceptions import StepRunnerException
+from ploigos_step_runner.results import StepResult
+from ploigos_step_runner.step_implementers.shared import ArgoCDGeneric
+
+DEFAULT_CONFIG = {
+    'argocd-add-or-update-target-cluster': True,
+    'argocd-cascade': True,
+    'argocd-propagation-policy': 'foreground',
+    'argocd-skip-tls': False,
+    'kube-api-skip-tls': False,
+    'kube-api-uri': 'https://kubernetes.default.svc'
+}
+
+REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS = [
+    'application-name',
+    'argocd-api',
+    'argocd-cascade',
+    'argocd-password',
+    'argocd-propagation-policy',
+    'argocd-skip-tls',
+    'argocd-username',
+    'branch',
+    'kube-api-skip-tls',
+    'kube-api-uri',
+    'organization',
+    'service-name'
+]
+
+class ArgoCDDelete(ArgoCDGeneric):
+    """
+    `StepImplementer` for the ArgoCD delete step.
+    """
+    @staticmethod
+    def step_implementer_config_defaults():
+        """Getter for the StepImplementer's configuration defaults.
+
+        Returns
+        -------
+        dict
+            Default values to use for step configuration values.
+
+        Notes
+        -----
+        These are the lowest precedence configuration values.
+        """
+        return {**DEFAULT_CONFIG}
+
+    @staticmethod
+    def _required_config_or_result_keys():
+        """Getter for step configuration or previous step result artifacts that are required before
+        running this step.
+
+        See Also
+        --------
+        _validate_required_config_or_previous_step_result_artifact_keys
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys or previous step result artifacts
+            that are required before running the step.
+        """
+        return REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS
+
+    def _run_step(self):  # pylint: disable=too-many-locals
+        """
+        Runs the step implemented by this StepImplementer to delete a ArgoCD
+        application by name.
+
+        Returns
+        -------
+        StepResult
+            Object containing the dictionary results of this step.
+        """
+        step_result = StepResult.from_step_implementer(self)
+
+        try:
+            argocd_app_name = self._get_app_name()
+            step_result.add_artifact(
+                name = 'argocd-app-name',
+                value = argocd_app_name
+            )
+
+            print("Sign into ArgoCD")
+            self._argocd_sign_in(
+                argocd_api = self._get_argocd_api(),
+                username = self._get_argocd_username(),
+                password = self._get_argocd_password(),
+                insecure = self._get_argocd_skip_tls()
+            )
+
+            add_or_update_target_cluster = self.get_value('argocd-add-or-update-target-cluster')
```

Comment:
we shouldn't ened to add a cluster if we are doing a delete should we?

Reply:
Sure? :grimacing: Logic removed!